pkg/tests/mappings/filter_directory/test_filter_directory.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Negative tests for the filter_directory script
+
+For a discussion as for why this needs to exist, see """
+
+import pathlib
+import os
+import sys
+import tempfile
+import unittest
+from rules_python.python.runfiles import runfiles
+
+# Get the filter_directory script into the import path.  There might be a
+# better way to do this, but it works.
+
+rf = runfiles.Create()
+filter_directory_script = rf.Rlocation(os.path.join(
+    os.environ["TEST_WORKSPACE"],
+    "filter_directory"
+))
+
+sys.path.append(os.path.dirname(filter_directory_script))
+
+import filter_directory
+
+
+# TODO: These tests are largely to ensure that filter_directory fails, but it
+# does not check _why_ they fail.
+#
+# This would involve changing how filter_directory returns errors, or maybe
+# restructuring it to make it unit-testable.
+#
+# Regardless, this would be significantly more code, and is not yet attempted.
+class FilterDirectoryInternalTest(unittest.TestCase):
+
+    def setUp(self):
+        self.indir = tempfile.TemporaryDirectory(dir=os.environ["TEST_TMPDIR"])
+        self.outdir = tempfile.TemporaryDirectory(dir=os.environ["TEST_TMPDIR"])
+        indir_path = pathlib.Path(self.indir.name)
+
+        (indir_path / "root").mkdir()
+        (indir_path / "root" / "a").open(mode='w').close()
+        (indir_path / "root" / "b").open(mode='w').close()
+        (indir_path / "root" / "subdir").mkdir()
+        (indir_path / "root" / "subdir" / "c").open(mode='w').close()
+        (indir_path / "root" / "subdir" / "d").open(mode='w').close()
+
+    def tearDown(self):
+        self.indir.cleanup()
+        self.outdir.cleanup()
+
+    ###########################################################################
+    # Test helpers
+    ###########################################################################
+    def callFilterDirectory(self,
```

Comment:
This is a strange indent style. I've never seen us put a ): alone, indented at 'def' spaces in any of the code here.
Usually we just would write  .... exclusions=None):

Reply:
I was trying to allow for trailing commas at the end of the arg list.  `):` has been moved to the same line as the last arg.